cognite/client/data_classes/data_modeling/shared.py

```diff
@@ -0,0 +1,94 @@
+from __future__ import annotations
+
+import typing
+from dataclasses import asdict, dataclass
+from typing import Any, Literal, Optional
+
+from cognite.client.utils._text import convert_all_keys_to_camel_case_nested, convert_all_keys_to_snake_case
+
+PrimitiveType = Literal["boolean", "float32", "float64", "int32", "int64", "timestamp", "date", "json"]
+CDFType = Literal["timeseries", "file", "sequence"]
+TextType = Literal["text"]
+DirectType = Literal["direct"]
+
+PRIMITIVE_TYPE_SET = set(typing.get_args(PrimitiveType))
+CDF_TYPE_SET = set(typing.get_args(CDFType))
+TEXT_TYPE_SET = set(typing.get_args(TextType))
+DIRECT_TYPE = set(typing.get_args(DirectType))
```

Comment:
These will be redundant when modeling property types with a proper type hierarchy.

Reply:
I still found use for these even with the new type hierarchy.